### PR TITLE
Improve backend code

### PR DESCRIPTION
**Pull request by thibaultleouay**

### CODE REVIEW

File: packages/api/src/router/incident.ts [hunk 1]
```diff
@@ -18,36 +18,29 @@ import {
 } from "@openstatus/db/src/schema";
 
 import { createTRPCRouter, protectedProcedure } from "../trpc";
+import {
+  hasUserAccessToWorkspace as getUserAndWorkspace,
+  hasUserAccessToWorkspace,
```
Comment (mxkaske) [inline packages/api/src/router/incident.ts]: Why having both?

File: packages/api/src/router/incident.ts [hunk 2]
```diff
@@ -18,36 +18,29 @@ import {
...
-        .select()
-        .from(usersToWorkspaces)
-        .where(eq(usersToWorkspaces.workspaceId, currentWorkspace.id))
-        .innerJoin(currentUser, eq(usersToWorkspaces.userId, currentUser.id))
-        .get();
-
+      const result = await hasUserAccessToWorkspace({
+        workspaceSlug: opts.input.workspaceSlug,
+        tenantId: opts.ctx.auth.userId,
+        ctx: opts.ctx,
```
Comment (mxkaske) [inline packages/api/src/router/incident.ts]: Can we **only** pass the prop `opts`? At least the `ctx` as auth is always included. No need to pass `tenantId` as extra property.

File: packages/api/src/router/utils.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * @returns {Promise<{ workspaceId: string; userId: string }> | undefined} - workspaceId and userId
+ */
+export const hasUserAccessToWorkspace = async ({
+  workspaceSlug,
+  tenantId,
+  ctx,
+}: {
+  workspaceSlug: string;
+  tenantId: string;
+  ctx: Context;
```
Comment (mxkaske) [inline packages/api/src/router/utils.ts:18]: Here e.g. we can access tenantId by `ctx.auth.tenantId`, no need to pass it extra.

File: packages/api/src/router/utils.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    .where(eq(workspace.slug, workspaceSlug))
+    .get();
+  if (!currentWorkspace) return;
+  const result = await ctx.db
+    .select()
+    .from(usersToWorkspaces)
+    .where(eq(usersToWorkspaces.workspaceId, currentWorkspace.id))
+    .innerJoin(currentUser, eq(usersToWorkspaces.userId, currentUser.id))
+    .get();
+  // the user don't have access to this workspace
```
Comment (mxkaske) [inline packages/api/src/router/utils.ts]:
```suggestion
  // the user doesn't have access to this workspace
```

**Review by mxkaske — COMMENTED** [review]
Good stuff 🔥

File: packages/api/src/router/incident.ts (hunk 1 above)
Comment (thibaultleouay) [reply packages/api/src/router/incident.ts]: Issue will be fixed 🤣

**Review by mxkaske — COMMENTED** [review]
lfg! 🚀

**Pull request merged**